packages/legend-application-studio/style/components/editor/mapping-editor/_property-mapping-editor.scss

```diff
@@ -37,14 +37,30 @@
     margin-left: 0.5rem;
   }
 
-  &__name {
+  &__name-with-validation {
     @include flexVCenter;
 
     height: 2.8rem;
     background: var(--color-dark-grey-300);
-    color: var(--color-light-grey-100);
     padding: 0 1rem;
     border-radius: 0.2rem;
+
+    &__name {
+      margin-left: 0.5rem;
+      margin-right: 0.5rem;
+      color: var(--color-light-grey-100);
+    }
+
+    &__error {
+      @include flexVCenter;
+
+      margin-left: 0.5rem;
+      color: var(--color-red-180);
+    }
+  }
+
+  &__name-with-validation-error {
+    background: var(--color-red-500);
```

Comment:
Maybe __label instead of __name but I'm not very sure? Also very minor but I believe modifiers in BEM convention structure are illustrated with two dashes (`--with-validation`)